build-tools/build-indexes

```diff
@@ -1017,7 +1016,216 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 			}
 		}
 	}
-
+	//Scoopapa code
+	console.log('');
+	BattleTeambuilderTable.ClientMods = {};
+	for (const modid of Object.keys(ClientMods)) {
+		try { //catch loading errors so the whole thing doesn't break if one mod does
+			//Gather data about formats
+			BattleTeambuilderTable.ClientMods[modid] = {};
+			const modDex = Dex.mod(modid);
+			const modData = modDex.data;
+			let modGen = 0;
+			BattleTeambuilderTable.ClientMods[modid].formats = [];
+			for (let formatid in Dex.data.Formats) { //Find formats using this mod
+				let format = Dex.getFormat(formatid);
+				if (format.mod === modid) {
+					if (formatid.slice(0, 3) === 'gen' && formatid.charAt(3) !== '8') modGen = (formatid.slice(0, 4));
+					BattleTeambuilderTable.ClientMods[modid].formats.push(format.fullname);
+				}
+			}
+			const standardTiers = ['uber', 'ou', 'uubl', 'uu', 'rubl', 'ru', 'nubl', 'nu', 'publ',
+				'pu', 'zu', 'nfe', 'lcuber', 'lc', 'cap', 'caplc', 'capnfe', 'ag', 'duber', 'dou',
+				'dbl', 'duu', 'dnu', 'illegal', 'unreleased'];
+			if (ClientMods[modid].customTiers) BattleTeambuilderTable.ClientMods[modid].customTiers = ClientMods[modid].customTiers;
+			else BattleTeambuilderTable.ClientMods[modid].customTiers = [];
+			for (const speciesid in modData.FormatsData) { //Find any nonstandard tiers used by this mod
+				const tier = modData.FormatsData[speciesid].tier;
+				if (typeof tier === 'undefined') continue;
+				if (!standardTiers.includes(toID(tier)) &&
+					!BattleTeambuilderTable.ClientMods[modid].customTiers.includes(tier)) {
+					BattleTeambuilderTable.ClientMods[modid].customTiers.push(tier);
+				}
+			}
+			BattleTeambuilderTable[modid] = {};
+			//tiers
+			const tierTable = {};
+			const pokemon = Object.keys(modData.Pokedex);
+			pokemon.sort();
+			const overrideTier = {};
+			BattleTeambuilderTable[modid].overrideTier = overrideTier;
+			for (const id of pokemon) {
+				const species = modDex.getSpecies(id);
+				const tier = species.tier;
+				overrideTier[species.id] = tier;
+				if (!tierTable[tier]) tierTable[tier] = [];
+				tierTable[tier].push(id);
+			}
+			const tiers = [];
+			const formatSlices = {};
+			BattleTeambuilderTable[modid].tiers = tiers;
+			BattleTeambuilderTable[modid].formatSlices = formatSlices;
+			const modTiers = BattleTeambuilderTable.ClientMods[modid].customTiers;
+			let tierOrder = ["OU", "AG", "Uber", "(Uber)", "OU", "(OU)", "UUBL", "UU", "RUBL", "RU", "NUBL", "NU", "PUBL", "PU", "(PU)", "New", "NFE", "LC Uber", "LC", "Unreleased"];
+			if (typeof modTiers !== "undefined") {
+				if (ClientMods[modid].excludeStandardTiers) tierOrder = modTiers;
+				else tierOrder = modTiers.concat(tierOrder);
+			}
+			for (const tier of tierOrder) {
+				if (tier in {OU:1, AG:1, Uber:1, UU:1, "(UU)":1, RU:1, NU:1, "(NU)":1, PU:1, "(PU)":1, NFE:1, LC:1, DOU:1, DUU:1, "(DUU)":1, New:1, Legal:1, Regular:1, "Restricted Legendary":1, "CAP LC":1}) {
```

Comment:
This should be 1) an array or Set, and 2) a constant variable

Reply:
you want me to make tierOrder a const?